README.md

```diff
@@ -1,171 +1,577 @@
+### `With` (multiple entities)
+The following examples apply to when you're seeding multiple entities (i.e with `SeedMany`).
 
+#### Set a property to a single value for all entities:
 ```csharp
-    public class OrderSeed : SeedFromDatabase<Order>, IDependsOn<Product>
-    {
-        private Product[] Products { get; set; }
+new BookingSeed().With(b => b.Name, "Booking Name");
+```
+> [!NOTE]
+> In this scenario, all entities will have the same value for the property.
 
-        public override Order GetSingle()
-        {
-            // Intialise a list of products from the database when null
-            // This list will be persisted until the seed process has completed
-            Products ??= DbEntity<Product>()
-                .Where(p => p.Deleted == null)
-                .Where(p => p.Price > 100)
-                .ToArray();
+#### Set a property to multiple values in order:
+```csharp
+new BookingSeed().With(b => b.Name, "Booking A", "Booking B", "Booking C");
+```
+
+#### Set a property based on the index of the entity being seeded
+```csharp
+new CouponSeed().With(c => c.Discount, index => index * 10m);
+```
+
+#### Set a property based on the previous entity:
+```csharp
+new CouponSeed().With(c => c.Discount, (_, previous) => previous?.Discount * 2 ?? 0.1m)
+```
 
-            var order = new Order();
+#### Set a property with a new value each time:
+```csharp
+new CouponSeed().With(
+    scs => scs.Name,
+    // Delegate is re-evaluated each time
+    () => Guid.NewGuid().ToString())
+```
+### `Without`
 
-            // Get 10 random products and add to order
-            var randomProducts = Random.Elements(Products, 10);
-            order.AddRange(randomProducts);
+#### Set a property on an entity as `null`;
+```csharp
+new BookingSeed().Without(b => b.Name);
+```
 
-            return order;
-        }
-    }
+
+### `WithNew` (single entity)
+Specify that an navigation optional property should be non-null, or override the default behaviour for a mandatory navigation property.
+#### Set a navigation property without a seed:
+```csharp
+new BookingSeed().WithNew(b => b.Coupon);
+```
+In this scenario, we will look for inheritors of `EntitySeed<Coupon>`, falling back on the default behaviour if none are found (see [Seeding without a class](#seeding-without-a-class)).
+
+#### Set a navigation property with a seed
+```csharp
+new BookingSeed().WithNew(b => b.Coupon, new CouponSeed().With(...));
+```
+
+#### Seed a grandparent entity
+```csharp
+new BookingSeed().WithNew(b => b.Facility.Building);
+```
+
+> [!NOTE]
+> This isn't necessary if these relationships are mandatory.
+
+If any parent in the property chain is optional, you will get a null reference exception. You can work around this as follows:
+```csharp
+new BookingSeed()
+    .WithNew(b => b.Parent.OptionalParent)// This is necessary to ensure `OptionalParent` is not null before seeding further properties on it.
```

Comment:
Would this example still make the point if we changed `b.Parent.OptionalParent` to simply `b.OptionalParent` for both lines?

Reply:
yes, will update as that's more succinct